src/async_op.c

```diff
@@ -35,30 +41,69 @@ static void async_op_dispose(ASYNC_OP* async_op)
 #define IS_POWER_OF_2(x) \
     (((x) & ((x) - 1)) == 0)
 
+/*tiny function that returns the next multiple of "align" starting from "start" (included). Does 1 mod and 1 branch*/
+static void* get_next_aligned_address(void* start, uint32_t align)
+{
+    void* result;
+    uint32_t m = (uintptr_t)start % align;
+    if (m == 0)
+    {
+        result = start;
+    }
+    else
+    {
+        result = (void*)((uintptr_t)start - m + align);
+    }
+    return result;
+}
+
+
+static void* get_prev_aligned_with_size_address(void* start, uint32_t size, uint32_t align)
+{
+    uintptr_t first_prev_address = (uintptr_t)start - size;
+    uint32_t m = first_prev_address % align;
+    if (m == 0)
+    {
+        return (void*)first_prev_address;
+    }
+    else
+    {
+        return (void*)((uintptr_t)first_prev_address - align);
+    }

```

Comment:
Consider replacing with return `(void*)((uintptr_t)first_prev_address - m ? align : 0)`

Reply:
I did quite a few variations of the same construct and looked in ASM what was generated. Line 44 sort of points that out.

This computation can be done in multiple way (including 0 branches) [where multiplication is used]

I've decided to keep it sort of human-readable also because of maintainability.

But I like your idea too and I think "with comments" can be nicer.

Reply:
hmmm... i don't think the calculation is right (mine or yours)